addons/payment/models/account_payment.py

```diff
@@ -53,6 +78,19 @@ def _compute_use_electronic_payment_method(self):
             codes = [key for key in dict(self.env['payment.acquirer']._fields['provider']._description_selection(self.env))]
             payment.use_electronic_payment_method = payment.payment_method_code in codes
 
+    def _compute_refunds_count(self):
+        rg_data = self.env['account.payment'].read_group(
+            domain=[
+                ('source_payment_id', 'in', self.ids),
+                ('payment_transaction_id.operation', '=', 'refund')
```

Comment:
What are the weight and the gain from an auto_join=True?

Reply:
After a talk together, we decided to put the auto_join, as it'll make queries faster.

The downside of auto_join is that if someone has access to the first table (`account.payment`), the query will proceed even if the user doesn't have access to the second table (`payment.transaction`). Since the ir.rule `payment_transaction_billing_rule` assure us that if someone has access to the payment, they have access to the transaction anyway, this isn't an issue here.